src/plot/formats.typ

```diff
@@ -0,0 +1,119 @@
+// Compare two floats
+#let _compare(a, b, eps: 1e-6) = {
+  return calc.abs(a - b) <= eps
+}
+
+// Pre-computed table of fractions
+#let _common-denoms = range(2, 11 + 1).map(d => {
```

Comment:
Idk. about this. It is the simplest solution I came up with and I think if you really want other fractions, it is fine, requiring specifying a fixed denominator.

Reply:
Is there a reason for `11 + 1` instead of `12`?

Reply:
To highlight that, we generate fractions up to `n/11`, not 12 (because range being zero-based).